### ISSUE TITLE
Autoselection does not work when adding the first tab with addComponentAtIndex

### ISSUE THREAD

**Issue by jm2rodriguez** [vaadin/vaadin-tabs-flow#102]
Programmatic selection of the first tab doesn't work when using Tabs.add(Tab... tabs)

**Code**
```java
Tabs tabs = new Tabs();
Tab firstTab = new Tab("First Tab");
Tab secondTab = new Tab("Second Tab");
tabs.add(firstTab, secondTab );
tabs.addSelectedChangeListener(new ComponentEventListener<Tabs.SelectedChangeEvent>() {...});
tabs.setSelectedIndex(0);   // First Tab isn't selected! SelectedChangeEvent is not fired
```

**Workaround**
The workaround is to use the method tabs.addComponentAtIndex(int index, Component component) instead of the method tabs.add(Tab... tabs).

```java
Tabs tabs = new Tabs();
Tab firstTab = new Tab("First Tab");
Tab secondTab = new Tab("Second Tab");
tabs.addComponentAtIndex(0, firstTab);
tabs.addComponentAtIndex(1, secondTab);
tabs.addSelectedChangeListener(new ComponentEventListener<Tabs.SelectedChangeEvent>() {...});
tabs.setSelectedIndex(0);   // First tab is selected. SelectedChangeEvent is fired
```

**Comment by pekam** [comment]
I tried your code, and for me the first tab is selected. But you are right in the fact that it doesn't fire a `SelectedChangeEvent`. It's not fired because `Tabs` makes the first tab automatically selected. So the selection happens as soon as the first tab is added, but you add the listener after it. By adding the listener before adding the tabs you can catch the first selection event. The line `setSelectedIndex(0)` is also not doing anything, as the first tab is already selected. The automatic selection of the first tab can optionally be disabled with `new Tabs(false)`.

But there is also a bug, as your second approach with `addComponentAtIndex` should behave exactly the same way as the first one. The issue seems to be that when using `addComponentAtIndex`, automatic selection of the first tab does not work. This is why in your example, calling `setSelectedIndex(0)` actually changes the selection and fires the event, as the first tab wasn't already selected. I don't consider this as a major issue, since `addComponentAtIndex` is mostly used after at least one tab is already added, and a new tab needs to be added in the beginning or in-between (otherwise the simpler `add`-method does the job).

I hope this helps to clarify the component's behavior. I will update the issue title if you don't mind.

**Comment by jm2rodriguez** [comment]
Thank you for this clarification

**Comment by jm2rodriguez** [comment]
Thank you for this clarification

**Comment by pekam** [comment]
Fixed in #108
